### PR TITLE
Add command to open current project

### PR DESCRIPTION
**Pull request by marcomorain**
Running `circleci open` will open the current git project on
circleci.com

### CODE REVIEW

File: cmd/open.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/CircleCI-Public/circleci-cli/git"
+	"github.com/spf13/cobra"
+)
+
+func openBrowser(url string) error {
```
Comment (glenjamin) [inline cmd/open.go]: Perhaps this? https://godoc.org/github.com/pkg/browser

A quick skim of the code suggests it's not obviously better, but `pkg` is generally a trusted source

File: git/git.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	Bitbucket VcsType = "BITBUCKET"
+)
+
+type Remote struct {
+	VcsType      VcsType
+	Organization string
+	Project      string
+}
+
+func InferOrganizationFromGitRemotes() (*Remote, error) {
```
Comment (glenjamin) [inline git/git.go]:
```suggestion
func InferProjectFromGitRemotes() (*Remote, error) {
```

File: cmd/open.go (hunk 1 above)
Comment (marcomorain) [reply cmd/open.go]: good idea

File: cmd/open.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/CircleCI-Public/circleci-cli/git"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+func projectUrl(remote *git.Remote) string {
+	return fmt.Sprintf("https://app.circleci.com/%s/%s/%s/pipelines",
```
Comment (glenjamin) [inline cmd/open.go]: does go have a url templating function? I suspect these values should be safe given that they came from the git URL - but we should probably be using some equivalent of `encodeURIComponent` on these path components when assembling a URL

File: cmd/open.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		remote.Organization,
+		remote.Project)
+}
+
+func openProjectInBrowser() error {
+
+	remote, err := git.InferProjectFromGitRemotes()
+
+	if err != nil {
+		return err
```
Comment (glenjamin) [inline cmd/open.go]: Are these errors human-friendly enough? should we be wrapping here with something nicer?

File: cmd/open.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+	"github.com/CircleCI-Public/circleci-cli/git"
+	"github.com/jtacoma/uritemplates"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+func projectUrl(remote *git.Remote) (string, error) {
+
+	template, err := uritemplates.Parse("https://app.circleci.com/{vcs}/{org}/{project}/pipelines")
```
Comment (glenjamin) [inline cmd/open.go]: This lib looks a bit old, I dunno how else to tell if it's a decent choice.

`url.PathEscape` might be a safer bet

Comment (marcomorain) [reply cmd/open.go]: I'd equate old with "stable" in this case – it's a small library that does a straightforward task.

File: cmd/open.go (hunk 4 above)
Comment (marcomorain) [reply cmd/open.go]: I've upgraded the error reporting:

```
Unable detect which URL should be opened. This command is intended to be run from
a git repository with a remote named 'origin' that is hosted on Github or Bitbucket
Error: This command must be run from inside a git repository
```

File: cmd/open.go (hunk 5 above)
Comment (danmux) [reply cmd/open.go]: Is the new dependancy, values map and extra error handling really better than..

```
func projectUrl(remote *git.Remote) string {
	return fmt.Sprintf("https://app.circleci.com/%s/%s/%s/pipelines",
		url.PathEscape(strings.ToLower(string(remote.VcsType))),
		url.PathEscape(remote.Organization),
		url.PathEscape(remote.Project))
}
```

File: cmd/open.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/CircleCI-Public/circleci-cli/git"
+	"github.com/jtacoma/uritemplates"
+	"github.com/pkg/browser"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func projectUrl(remote *git.Remote) (string, error) {
+
+	template, err := uritemplates.Parse("https://app.circleci.com/{vcs}/{org}/{project}/pipelines")
```
Comment (danmux) [inline cmd/open.go]: Is the new dependancy, values map and extra error handling really better than..

```
func projectUrl(remote *git.Remote) string {
	return fmt.Sprintf("https://app.circleci.com/%s/%s/%s/pipelines",
		url.PathEscape(strings.ToLower(string(remote.VcsType))),
		url.PathEscape(remote.Organization),
		url.PathEscape(remote.Project))
}
```

Comment (marcomorain) [reply cmd/open.go]: nope, I added it at @glenjamin 's suggestion. I've removed it.

**Pull request merged**